lib/src/facts/linux/virtualization_resolver.cc

```diff
@@ -25,6 +21,31 @@ namespace lth_file = leatherman::file_util;
 
 namespace facter { namespace facts { namespace linux {
 
+    string virtualization_resolver::get_cloud_provider(collection& facts)
+    {
+        // Check for Azure
+        std::string provider = get_azure(facts);
+
+        return provider;
+    }
+
+    string virtualization_resolver::get_azure(collection& facts, string const& leases_file)
+    {
+        std::string provider;
+        if ( boost::filesystem::exists( leases_file ) )
+        {
+            lth_file::each_line(leases_file, [&](string& line) {
+                if (line.find("option 245") != std::string::npos ||
```

Comment:
Is there anywhere that documents what this means and why it exists? It'd be nice to include a comment about why we search for this string.